cern_access/indico_cern_access/plugin.py

```diff
@@ -0,0 +1,238 @@
+    def _event_title_changed(self, event, changes, **kwargs):
+        """Updates event name in the ADaMS CERN access request"""
+        if 'title' not in changes:
+            return
+        requested_registrations = get_requested_registrations(event=event)
+        if requested_registrations:
+            state, _ = send_adams_post_request(event, requested_registrations, update=True)
+            if state == CERNAccessRequestState.accepted:
+                update_access_requests(requested_registrations, state)
+
+    def _is_ticketing_handled(self, regform):
+        """Checks if a registration form has requested access to CERN, if so the plugin will handle tickets mailing """
+        if regform.cern_access_request and regform.cern_access_request.is_active:
+            return True
+        return False
+
+    def _form_validated(self, form, **kwargs):
+        """Forbids to disable the tickets when access to CERN is requested and
+         to use CERN access ticket template with regforms without accepted CERN access request
+        """
+        if not isinstance(form, TicketsForm):
+            return
+        regform = RegistrationForm.get_one(request.view_args['reg_form_id'])
+        if regform.cern_access_request and regform.cern_access_request.is_active and form.tickets_enabled.data is False:
+            err = _('Access to CERN is requested for participants registered with this form, ticketing must be enabled')
+            form.tickets_enabled.errors.append(err)
+            return False
+        access_tpl = self.settings.get('access_ticket_template')
+        ticket_template = DesignerTemplate.get_one(form.ticket_template_id.data)
+        if not access_tpl:
+            return
+        if ticket_template == access_tpl or ticket_template.backside_template == access_tpl:
+            if (not regform.cern_access_request or
+                    (regform.cern_access_request and
+                        regform.cern_access_request.request_state != CERNAccessRequestState.accepted)):
+                form.ticket_template_id.errors.append(_('Selected template can only be used with an '
+                                                        'accepted CERN access request'))
+                return False
+
+    def _print_badge_template(self, template, regform, **kwargs):
+        access_tpl = self.settings.get('access_ticket_template')
+        if not access_tpl:
+            return
+        if template == access_tpl or template.backside_template == access_tpl:
+            if (not regform.cern_access_request or
+                    (regform.cern_access_request and
+                        regform.cern_access_request.request_state != CERNAccessRequestState.accepted)):
+                raise Forbidden(_('This badge cannot be printed because it uses the CERN access ticket '
+                                  'template without an accepted CERN access request'))
+
+    def _generate_ticket_qr_code(self, registration, ticket_data, **kwargs):
```

Comment:
:warning: Shouldn't there be a check in here to only add the extra data if the regform is actually used for CERN access?!